test/connection_test.py

```diff
@@ -0,0 +1,138 @@
+# Databricks notebook source
+from data_io.connection import build_connection
+from urlpath import URL
+from pyspark.sql import SparkSession
+from pyspark.dbutils import DBUtils
+"""
+this class is for test mining the credentials
+"""
+
+class GetConnection():
+  
+  def __init__(self):
+    spark = SparkSession.builder.getOrCreate()
+    dbutils = DBUtils(spark)
+    self.scope = (
+                dbutils
+                .notebook
+                .entry_point
+                .getDbutils()
+                .notebook()
+                .getContext()
+                .tags()
+                .apply('user')
+                # XXX Hack incoming
+                .split('@')
+                [0]
+            )
+  # get secret
+  def get_secret(self, key):
+    spark = SparkSession.builder.getOrCreate()
+    dbutils = DBUtils(spark)
+    return dbutils.secrets.get(scope=self.scope, key=key)
+    
+  # providing list of creds
+  def get_connections(self):
+    list_creds = []
+    secret = self.get_secret('credentials.cfg')
+    connection_list = []
```

Comment:
The credentials file is a standardized INI format. There are standard parsers to do what you're doing here.

Example:
https://github.com/chris-bishop-tfs/data_io/blob/main/data_io/connection.py#L316-L317

Reply:
The parser was not working in databrick. Where unittest are meant to be done for databrick packages

Reply:
What does "not working" mean?

Reply:
Question still open.